src/tests/secp256k1_blake160_sighash_all.rs

```diff
@@ -240,5 +233,136 @@ fn test_super_long_witness() {
+            keys.push(key_gen.random_privkey());
+        }
+
+        keys
+    }
+
+    fn pkh(key: &Privkey) -> Bytes {
+        Bytes::from(blake160(&key.pubkey().unwrap().serialize()).as_bytes())
+    }
+
+    fn verify(data_loader: &DummyDataLoader, tx: &Transaction) -> Result<u64, ScriptError> {
+        let resolved_tx = build_resolved_tx(&data_loader, &tx);
+        let script_config = ScriptConfig::default();
+        TransactionScriptsVerifier::new(&resolved_tx, data_loader, &script_config)
+            .verify(MAX_CYCLES)
+    }
+
+    #[test]
+    fn test_sighash_1_2_unlock() {
+        let mut data_loader = DummyDataLoader::new();
+        let keys = generate_keys(3);
+
+        let raw_tx = gen_tx(
+            &mut data_loader,
+            SIGHASH_ALL_BIN.clone(),
+            vec![pkh(&keys[0]), pkh(&keys[1]), vec![1].into()],
+        );
+
+        {
+            let tx = sign_tx(raw_tx.clone(), &keys[0]);
+            verify(&data_loader, &tx).expect("pass verification");
+        }
+        {
+            let tx = sign_tx(raw_tx.clone(), &keys[1]);
+            verify(&data_loader, &tx).expect("pass verification");
+        }
+        {
+            let tx = sign_tx(raw_tx.clone(), &keys[2]);
+            assert_eq!(
+                verify(&data_loader, &tx),
+                Err(ScriptError::ValidationFailure(ERROR_PUBKEY_BLAKE160_HASH))
+            );
+        }
+    }
+
+    #[test]
+    fn test_sighash_2_3_unlock() {
+        let mut data_loader = DummyDataLoader::new();
+        let keys = generate_keys(4);
+
+        let raw_tx = gen_tx(
+            &mut data_loader,
+            SIGHASH_ALL_BIN.clone(),
+            vec![pkh(&keys[0]), pkh(&keys[1]), pkh(&keys[2]), vec![2].into()],
+        );
+
+        {
+            let tx = multi_sign_tx(raw_tx.clone(), &[&keys[0], &keys[1]]);
+            verify(&data_loader, &tx).expect("pass verification");
+        }
+        {
+            let tx = multi_sign_tx(raw_tx.clone(), &[&keys[1], &keys[0]]);
+            verify(&data_loader, &tx).expect("pass verification");
+        }
+        {
+            let tx = multi_sign_tx(raw_tx.clone(), &[&keys[0], &keys[2]]);
+            verify(&data_loader, &tx).expect("pass verification");
+        }
+        {
+            let tx = multi_sign_tx(raw_tx.clone(), &[&keys[1], &keys[2]]);
+            verify(&data_loader, &tx).expect("pass verification");
+        }
+
+        {
+            let tx = sign_tx(raw_tx.clone(), &keys[1]);
+            assert_eq!(
+                verify(&data_loader, &tx),
+                Err(ScriptError::ValidationFailure(
+                    ERROR_WRONG_NUMBER_OF_ARGUMENTS
+                ))
+            );
+        }
+        {
+            let tx = multi_sign_tx(raw_tx.clone(), &[&keys[0], &keys[1], &keys[2]]);
+            assert_eq!(
+                verify(&data_loader, &tx),
+                Err(ScriptError::ValidationFailure(
+                    ERROR_WRONG_NUMBER_OF_ARGUMENTS
```

Comment:
If more than 2 signatures, it is failed also?

Reply:
The current script allows extra witness args, I'm not sure we have to add this limit when adding the multisig. cc @xxuejie

Reply:
I prefer to allow extra witness args since this is the default lock script, and the extra args might have more uses(NerbosDAO for example)